src/ngx_http_lua_args.c

```diff
@@ -40,10 +73,14 @@ ngx_http_lua_ngx_req_set_uri_args(lua_State *L)
 
     ngx_http_lua_check_fake_request(L, r);
 
-    switch (lua_type(L, 1)) {
+    type = lua_type(L, 1);
+    switch (type) {
     case LUA_TNUMBER:
     case LUA_TSTRING:
         p = (u_char *) lua_tolstring(L, 1, &len);
+        if (type == LUA_TSTRING && !ngx_http_lua_valid_args(p, len)) {
+            return luaL_argerror(L, 1, "args need to be escaped");
```

Comment:
We should print out offending byte otherwise it would be very difficult to debug such errors. Also, error messages should be saying what is wrong instead of only what to do. So a message like "query-string contains unescaped byte 0x09" would be much better here.